rundeckapp/src/test/groovy/org/rundeck/app/providers/GormUserDataProviderSpec.groovy

```diff
@@ -289,16 +289,24 @@ class GormUserDataProviderSpec extends RundeckHibernateSpec implements DataTest
     @Unroll
     def "Should validate user exists"() {
         given:
-        User savedUser = new User(login: "saved")
-        savedUser.save()
+        User savedUser = new User(login: login)
+        if(persitUser){
+            savedUser.save()
+        }
+        provider.featureService = Mock(FeatureService){
+            1 * featurePresent(Features.CASE_INSENSITIVE_USERNAME) >> caseSensitive
+        }
         when:
-        Boolean userExists = provider.validateUserExists(login)
+        Boolean userExists = provider.validateUserExists(login.toUpperCase())
         then:
         userExists == expect
         where:
-        login   | expect
-        "login" | false
-        "saved" | true
+        login   | expect   | caseSensitive | persitUser
```

Comment:
nitpick: `caseSensitive` should be called `caseInsensitive` since the meaning of the feature flag has been inverted.